inference-engine/tests/functional/plugin/shared/include/behavior/preprocessing.hpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <vector>
+
+#include <ie_core.hpp>
+#include "common_test_utils/test_assertions.hpp"
+#include "common_test_utils/common_utils.hpp"
+#include "functional_test_utils/plugin_cache.hpp"
+#include "functional_test_utils/layer_test_utils.hpp"
+#include "functional_test_utils/blob_utils.hpp"
+#include "ie_preprocess.hpp"
+#include "functional_test_utils/behavior_test_utils.hpp"
+#include <ngraph/pass/visualize_tree.hpp>
+
+namespace {
+void setInputNetworkPrecision(InferenceEngine::CNNNetwork &network, InferenceEngine::InputsDataMap &inputs_info,
+                              InferenceEngine::Precision input_precision) {
+    inputs_info = network.getInputsInfo();
+    ASSERT_EQ(1u, inputs_info.size());
+    inputs_info.begin()->second->setPrecision(input_precision);
+}
+
+}
+
+namespace BehaviorTestsDefinitions {
+
+using  PreprocessingPrecisionConvertParams = std::tuple<
+        InferenceEngine::Precision,         // Input precision
+        std::string,                        // Device name
+        std::map<std::string, std::string>  // Config
+>;
+
+struct PreprocessingPrecisionConvertTest :
+        public testing::WithParamInterface<PreprocessingPrecisionConvertParams>,
+        LayerTestsUtils::LayerTestsCommon {
+public:
+    static std::string getTestCaseName(testing::TestParamInfo<PreprocessingPrecisionConvertParams> obj) {
+        InferenceEngine::Precision  inPrc;
+        std::string targetDevice;
+        std::map<std::string, std::string> configuration;
+        std::tie(inPrc, targetDevice, configuration) = obj.param;
+        std::ostringstream result;
+        result << "inPRC=" << inPrc.name() << "_";
+        result << "targetDevice=" << targetDevice;
+        if (!configuration.empty()) {
+            for (auto& configItem : configuration) {
+                result << "configItem=" << configItem.first << "_" << configItem.second << "_";
+            }
+        }
+        return result.str();
+    }
+
+    void SetUp() override {
+        SetRefMode(LayerTestsUtils::RefMode::INTERPRETER);
+
+        std::tie(inPrc, targetDevice, configuration) = this->GetParam();
+
+        bool specialZero = true;
+
+        std::vector<size_t> inputShape    {4, 4};
+
+        //Idea of the tests is
```

Comment:
Minor corrections (please check the 2nd bullet! It wasn't readable, but my fixes might've ruined the meaning): 
```
// This test:
// - Uses a no-op graph to test the plugin internal precision preprocessing only.
// - Mimics the reference code preprocessing via an extra nGraph Convert operation.
// Two graphs are created: a graph for the plugin and a graph for the nGraph reference code. 
```
Also in the last one - who exactly creates the graphs? Some particular function? If so, let's state it

Reply:
> 
> 
> Minor corrections (please check the 2nd bullet! It wasn't readable, but my fixes might've ruined the meaning):
> 
> ```
> // This test:
> // - Uses a no-op graph to test the plugin internal precision preprocessing only.
> // - Mimics the reference code preprocessing via an extra nGraph Convert operation.
> // Two graphs are created: a graph for the plugin and a graph for the nGraph reference code. 
> ```
> 
changed to : 

        // This test:
        // - Strive to test the plugin internal preprocessing (precision conversion) only.
        //   Thus (logically) no-op graph is used.
        // - Reference code mimic the preprocessing via extra ngraph Convert operation.
        // - Create/uses two (different) graphs here : one to feed the the plugin and one calculate the reference result.

and moved the comment to the beginning of the function. 

> Also in the last one - who exactly creates the graphs? Some particular function? If so, let's state it
the graphs a re created via helper lambda : 
```cpp
        function            = make_ngraph(false); 
        reference_function  = make_ngraph(true);  //use extra ops to mimic the preprocessing
```
Does it make more sense now ?